### ISSUE TITLE
SELECT times out in test_simple_kill_new_node_while_bootstrapping_with_parallel_writes_in_multidc

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#10094]
test_simple_kill_new_node_while_bootstrapping_with_parallel_writes_in_multidc seems flaky since
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/1088/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/test_simple_kill_new_node_while_bootstrapping_with_parallel_writes_in_multidc/
Scylla version 3e31126bdf90bcca076367a510ae2362f741cdbb
```
Stacktrace
self = <update_cluster_layout_tests.TestUpdateClusterLayout object at 0x7fab2aeb73d0>

    def test_simple_kill_new_node_while_bootstrapping_with_parallel_writes_in_multidc(self):
        """
        Test bootstrapped node streams all data
        1. Create a cluster with two nodes in multidc with rf=1, insert data
        2. Add node, wait for each to start bootstrapping and write additional data
        3. kill it
        4. Check that the cluster returns all
        """
        cluster = self.cluster
    
        # Disable hinted handoff and set batch commit log so this doesn't
        # interfer with the test (this must be after the populate)
        cluster.set_configuration_options(
            values=self.default_config_options(), batch_commitlog=True)
        cluster.populate([1, 1]).start(wait_for_binary_proto=True, wait_other_notice=True)
        node1 = cluster.nodelist()[0]
        node2 = cluster.nodelist()[1]
    
        session = self.cql_connection(node1)
        create_ks(session, 'ks', {'dc1': 1, 'dc2': 1})
        create_cf(session, 'cf', read_repair=0.0, columns={'c1': 'text', 'c2': 'text'})
    
        insert_c1c2(session, keys=range(1000), consistency=ConsistencyLevel.EACH_QUORUM)
    
        logger.debug("Inserting more data to make streaming process longer...")
        node1.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks1'])
        node2.stress(['write', 'n=5000', 'no-warmup', '-schema', 'replication(factor=3) keyspace=ks2'])
    
        # create a new node and adding it - we cannot do this more then once
        a_new_node = new_node(cluster, data_center='dc1')
        failed = before = None
    
        def run():
            try:
                logger.debug("start write")
                for key in range(2000, 4000):
                    # working around the default retry_policy that attempts 5 times
                    statement = SimpleStatement("INSERT INTO cf (key, c1, c2) VALUES ('k%d', 'value1', 'value2')" %
                                                key, consistency_level=ConsistencyLevel.EACH_QUORUM,
                                                retry_policy=FallthroughRetryPolicy())
                    tbefore = str(datetime.now())
                    session.execute(statement)
                logger.debug("end write")
                failed = 'insert should have failed'
            except (Unavailable) as e:
                tfailed = str(datetime.now())
                logger.debug("exception thrown Unavailable %s" % e)
                pass
            except (WriteTimeout) as e:
                tfailed = str(datetime.now())
                logger.debug("exception thrown WriteTimeout %s" % e)
                pass
            except (OperationTimedOut) as e:
                tfailed = str(datetime.now())
                failed = "Server side exception not thrown driver side exception thrown OperationTimeout %s %s %s" %\
                         (e, before, tfailed)
    
        executor = ThreadPoolExecutor(max_workers=1)
    
        logger.debug("Start Node")
        a_new_node.start(jvm_args=['--logger-log-level', 'stream_session=debug'], no_wait=True)
        a_new_node.watch_log_for("JOINING: Starting to bootstrap")
        time.sleep(1)
        t = executor.submit(run)
        time.sleep(1)
        a_new_node.watch_log_for("JOINING: Starting to bootstrap")
        a_new_node.watch_log_for("Beginning stream session|sync data for keyspace=ks, status=started")
        self.wait_for_nodes_status(node1, [['UN', 'UJ', 'UN'], ['UN', 'UN', 'UN']])
        logger.debug("Stop Node")
        a_new_node.stop(gently=False)
        t.result()
        assert failed is None
        self.wait_for_nodes_status(node1, [['UN', 'UN'], ['UN', 'DN', 'UN']])
    
        logger.debug("Query Again")
>       session.execute("SELECT * FROM cf")

update_cluster_layout_tests.py:568: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cassandra/cluster.py:2611: in cassandra.cluster.Session.execute
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   cassandra.OperationTimedOut: errors={'127.0.90.1:9042': 'Client request timeout. See Session.execute[_async](timeout)'}, last_host=127.0.90.1:9042

cassandra/cluster.py:4829: OperationTimedOut
```

**Comment by bhalevy** [comment]
It appears like the flakiness we saw was due to a test issue fixed in https://github.com/scylladb/scylla-dtest/pull/2609
where the bootstrapping node was stopped too late, after i has joined the cluster, causing the later select query to fail.
The tests pass reliably for with with https://github.com/scylladb/scylla-dtest/pull/2609 and in particular, https://github.com/scylladb/scylla-dtest/commit/0c7e09e91327e3089ddc86256800c321cce3ce8d,
that replaces the call to `wait_for_nodes_status` with a log regular expression that detects the node is in the middle of streaming / repair-based bootstrap for the test keyspace.

Let's close this issue once the dtest is verified to pass on dtest-release.
